storage-proofs/porep/src/stacked/vanilla/cache.rs

```diff
@@ -157,9 +157,21 @@ impl ParentCache {
     {
         let path = cache_path(cache_entries, graph);
         if path.exists() {
-            Self::open(len, cache_entries, graph, path)
+            Self::open(len, cache_entries, graph, &path)
         } else {
-            Self::generate(len, cache_entries, graph, path)
+            match Self::generate(len, cache_entries, graph, &path) {
+                Ok(c) => Ok(c),
```

Comment:
I think `.or_else()` could be used. Though I'm not sure if that makes things better.

Reply:
it made the code even harder to read